batchflow/batch.py

```diff
@@ -493,22 +494,73 @@ def apply_transform(self, ix, func, *args, src=None, dst=None, p=None, use_self=
 
             for item in range(len(batch)):
                 self.dst[item] = func(self.src[item], *args, **kwargs)
+
+        If `src` is a list with two or more elements, `dst` should be list or
+        tuple of the same lenght.
+
+        Examples
+        --------
+
+        ::
+
+            apply_transform(make_masks_fn, src='images', dst='masks')
+            apply_transform(apply_mask, src=('images', 'masks'), dst='images', use_self=True)
+            apply_transform_all(rotate, src=['images', 'masks'], dst=['images', 'masks'], p=.2)
         """
+        to_act = bool(p is None or np.random.binomial(1, p))
+
+        if isinstance(src, list) and len(src) > 1 and isinstance(dst, (list, tuple)) and len(src) == len(dst):
+            return tuple([self._apply_transform(ix, func, to_act, *args, src=src_component,
+                                                dst=dst_component, use_self=use_self, **kwargs)
+                          for src_component, dst_component in zip(src, dst)])
+        return self._apply_transform(ix, func, to_act, *args, src=src, dst=dst, use_self=use_self, **kwargs)
+
+    def _apply_transform(self, ix, func, to_act, *args, src=None, dst=None, use_self=False, **kwargs):
+        """ Apply a function to each item in the batch.
+
+        Parameters
+        ----------
+        func : callable
+            a function to apply to each item from the source
+
+        to_act: bool
```

Comment:
There is no point in changing method signature. Leave `p` as in `apply_transform`

Reply:
Changed it back

Reply:
Remove it from a docstring too